Tests/RSContextTests.swift

```diff
@@ -0,0 +1,248 @@
+        XCTAssertEqual(context.device.advertisingId, "some-random-uuid");
+        XCTAssertTrue(context.device.adTrackingEnabled)
+    }
+    
+    func test_updateExternalIds() {
+        context.updateExternalIds(NSMutableArray(array:externalIds1))
+        XCTAssertEqual(context.externalIds, NSMutableArray(array:externalIds1))
+        context.updateExternalIds(NSMutableArray(array:externalIds2))
+        XCTAssertEqual(context.externalIds, NSMutableArray(array:finalExternalIds))
+    }
+    
+    func test_persistExternalIds() {
+        context.updateExternalIds(NSMutableArray(array:externalIds1))
+        context.persistExternalIds()
+        XCTAssertEqual(preferenceManager.getExternalIds(), convertToJSONString(arrayObject: externalIds1))
+        context.updateExternalIds(NSMutableArray(array:externalIds2))
+        context.persistExternalIds()
+        XCTAssertEqual(preferenceManager.getExternalIds(), convertToJSONString(arrayObject: finalExternalIds))
+    }
+    
+    func test_resetExternalIds() {
+        context.updateExternalIds(NSMutableArray(array:externalIds1))
+        XCTAssertEqual(context.externalIds, NSMutableArray(array:externalIds1))
+        context.resetExternalIdsOnQueue()
+        XCTAssertNil(context.getExternalIds())
+        XCTAssertNil(preferenceManager.getExternalIds())
+        context.updateExternalIds(NSMutableArray(array:externalIds2))
+        XCTAssertEqual(context.externalIds, NSMutableArray(array:externalIds2))
+        context.resetExternalIdsOnQueue()
+        XCTAssertNil(context.getExternalIds())
+        XCTAssertNil(preferenceManager.getExternalIds())
+    }
+    
+    func test_putAppTrackingConsent() {
+        context.putAppTrackingConsent(-1)
+        XCTAssertEqual(context.device.attTrackingStatus, RSATTNotDetermined)
+        context.putAppTrackingConsent(0)
+        XCTAssertEqual(context.device.attTrackingStatus, RSATTNotDetermined)
+        context.putAppTrackingConsent(1)
+        XCTAssertEqual(context.device.attTrackingStatus, RSATTRestricted)
+        context.putAppTrackingConsent(2)
+        XCTAssertEqual(context.device.attTrackingStatus, RSATTDenied)
+        context.putAppTrackingConsent(3)
+        XCTAssertEqual(context.device.attTrackingStatus, RSATTAuthorize)
+        context.putAppTrackingConsent(4)
+        XCTAssertEqual(context.device.attTrackingStatus, RSATTAuthorize)
+    }
+    
+    func test_setSessionData() {
+        let userSession = RSUserSession.initiate(RSSessionInActivityDefaultTimeOut, with: preferenceManager)
+        userSession.start()
+        context.setSessionData(userSession)
+        XCTAssertEqual(context.sessionId, userSession.getId())
+        XCTAssertTrue(context.sessionStart)
+    }
+    
+    func test_setConsentData() {
+        var deniedConsentIds = ["deniedId1", "deniedId2", "deniedId3", "deniedId4"]
+        context.setConsentData(deniedConsentIds)
+        XCTAssertEqual(context.getConsentData(), ["deniedConsentIds": deniedConsentIds])
```

Comment:
Instead of getConsentData, persist the context object. You don't need this functions.

Reply:
fixed this now